airflow/providers/amazon/provider.yaml

```diff
@@ -128,6 +124,10 @@ additional-extras:
   - name: python3-saml
     dependencies:
       - python3-saml>=1.16.0
+      # XML sec 1.3.14 breaks Amazon's authentication with `lxml & xmlsec libxml2 library version mismatch`
+      # We should investigate if we can upgrade to a newer version of lxml and xmlsec
+      # Tracked in https://github.com/apache/airflow/issues/39103
+      - xmlsec<1.3.14
```

Comment:
This will create an extra dependency is it not?
if so then when we want to remove it (as it might not be needed in the future) it will require having a major release for the provider?

Reply:
`xmlsec` is a dependency of `python3-saml`, so it's always been there, and likely will always be.

The change 3 weeks ago was just to set an upper bound. This PR just moves it into the `python3-saml` extra.